client/src/components/JurisdictionAdmin/BatchInventory.tsx

```diff
@@ -0,0 +1,483 @@
+  const uploadFiles = useUploadFiles(key, url)
+  const deleteFile = useDeleteFile(key, url)
+  const queryClient = useQueryClient()
+  return {
+    uploadedFile: useUploadedFile(key, url, {
+      onFileChange: () =>
+        // Tabulator status file is reprocessed when CVR is uploaded
+        queryClient.invalidateQueries([
+          'batchInventory',
+          jurisdictionId,
+          'tabulatorStatus',
+        ]),
+    }),
+    uploadFiles: files => {
+      const formData = new FormData()
+      formData.append('cvr', files[0], files[0].name)
+      return uploadFiles.mutateAsync(formData)
+    },
+    uploadProgress: uploadFiles.progress,
+    deleteFile: () => deleteFile.mutateAsync(),
+    // TODO make this actually work
+    downloadFileUrl: `${url}/file`,
+  }
+}
+
+const useBatchInventoryTabulatorStatus = (
+  electionId: string,
+  jurisdictionId: string
+): IFileUpload => {
+  const key = ['batchInventory', jurisdictionId, 'tabulatorStatus']
+  const url = `/api/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/tabulator-status`
+  const uploadFiles = useUploadFiles(key, url)
+  const deleteFile = useDeleteFile(key, url)
+  return {
+    uploadedFile: useUploadedFile(key, url),
+    uploadFiles: files => {
+      const formData = new FormData()
+      formData.append('tabulatorStatus', files[0], files[0].name)
+      return uploadFiles.mutateAsync(formData)
+    },
+    uploadProgress: uploadFiles.progress,
+    deleteFile: () => deleteFile.mutateAsync(),
+    // TODO make this actually work
+    downloadFileUrl: `${url}/file`,
+  }
+}
+
+const useBatchInventorySignOff = (
+  electionId: string,
+  jurisdictionId: string
+) => {
+  const queryClient = useQueryClient()
+  const key = ['batchInventory', jurisdictionId, 'signOff']
+  const url = `/api/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/sign-off`
+  const status = useQuery<{ signedOffAt: string | null }>(key, () =>
+    fetchApi(url)
+  )
+  const signOff = useMutation(() => fetchApi(url, { method: 'POST' }), {
+    onSuccess: () => queryClient.invalidateQueries(key),
+  })
+  const undoSignOff = useMutation(() => fetchApi(url, { method: 'DELETE' }), {
+    onSuccess: () => queryClient.invalidateQueries(key),
+  })
+  return {
+    status,
+    signOff,
+    undoSignOff,
+  }
+}
+
+const STEPS = [
+  'Upload Election Results',
+  'Inventory Batches',
+  'Download Audit Files',
+] as const
+
+const StepContainer = styled.div`
+  background-color: ${Colors.LIGHT_GRAY5};
+`
+
+const StepPanel = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-around;
+  min-height: 300px;
+  padding: 50px 20px;
+`
+
+const StepActions: React.FC<{
+  left?: React.ReactElement
+  right?: React.ReactElement
+}> = ({ left, right }) => (
+  <div
+    style={{
+      display: 'flex',
+      padding: '20px',
+      backgroundColor: Colors.LIGHT_GRAY2,
+      justifyContent: 'space-between',
+    }}
+  >
```

Comment:
@arsalansufi I'm curious if you have thoughts about using styled-components vs inline styles. I just did whatever was quickest for prototyping, but would like to decide on a strategy and clean up accordingly

Reply:
Yes good q! In my work on the audit planner, I've been trying to consistently use styled components instead of inline styles. There are limitations to inline styles, e.g. lack of support of `:hover`, `:active`, `:focus` selectors, lack of support for media queries, less separation of style and structure. And while these limitations don't pose problems for simple cases, styled components are guaranteed to work for all cases. So if we are to recommend one over the other as a general rule, I'd pick styled components

Of course, I do get that it can be cumbersome when you just need to add a one-off margin to a button, for example

Reply:
Ok, that sounds good. I'll try moving in that direction.